### ISSUE TITLE
no-map-without-usage reports a false positive inside constructors

### ISSUE THREAD

**Issue by karol-majewski** [Glavin001/tslint-clean-code#53]
I believe the following should be accepted without an error:

```ts
const entities = [{ id: 0, name: "John" }];

new Map(entities.map<[number, string]>(entity => [entity.id, entity.name]));
```

**Comment by Glavin001** [comment]
Published to v0.2.7

Thank you for contributing!

**Comment by raphaklaus** [comment]
Not working properly if you're using `Promise.map`

**Comment by karol-majewski** [comment]
@raphaklaus The rule promises to work with to ECMAScript's `Array.prototype.map`, not any library-specific implementations. ;-)

**Comment by Glavin001** [comment]
@raphaklaus Pull Requests welcome! 😃
